### PR TITLE
Document Vault Default Password Policy

### PR DESCRIPTION
**Pull request by ram-parameswaran**
Document Vault Default Policy example for easy reference. Currently the definition is rather easter egged [here](https://developer.hashicorp.com/vault/api-docs/secret/databases#password_policy)

### CODE REVIEW

File: website/content/docs/concepts/password-policies.mdx [hunk 1]
```diff
@@ -360,6 +360,39 @@ This policy generates 8 character passwords from the charset `abcde01234` and re
 character from `01234` to be in it, but does not require any characters from `abcde`. The password
 `04031945` may result from this policy, even though no alphabetical characters are in it.
 
+## Default Password Policy
```
Comment (schavis) [inline website/content/docs/concepts/password-policies.mdx:363]:
```suggestion
## Default password policy
```
Style correction: use sentence case for headings

File: website/content/docs/concepts/password-policies.mdx [hunk 2]
```diff
@@ -360,6 +360,39 @@ This policy generates 8 character passwords from the charset `abcde01234` and re
...
+
+Vault ships with a Default Password Policy and this gets applied to any password that Vault 
+generates where a custom password policy is not explicitly set. The default password policy is 
+defined below.
```
Comment (schavis) [inline website/content/docs/concepts/password-policies.mdx:367]:
```suggestion
--------------------------------------------------------------------------------
Vault ships with a default password policy that applies to any password 
generated by Vault  without an explicit policy assignment. The default
policy requires passwords include:

- 20 characters total
- 1 uppercase character
- 1 lowercase character
- 1 number
- 1 special character
```
Style correction: reserve capitalization for products (not features/functionality), write in the active voice, avoid "this" as a pronoun.

File: website/content/docs/concepts/password-policies.mdx [hunk 3]
```diff
@@ -360,6 +360,39 @@ This policy generates 8 character passwords from the charset `abcde01234` and re
...
+
+<details>
+<summary>20 characters with at least 1 uppercase, 1 lowercase, 1 number, and 1 dash character.</summary>
```
Comment (schavis) [inline website/content/docs/concepts/password-policies.mdx:370]:
```suggestion
```
Folded into the previous section. Also, I'm not sure what the `details` and `summary` tags are for? I don't think the markdown parser recognizes those tags as valid formatting

File: website/content/docs/concepts/password-policies.mdx [hunk 4]
```diff
@@ -360,6 +360,39 @@ This policy generates 8 character passwords from the charset `abcde01234` and re
...
+  charset = "0123456789"
+  min-chars = 1
+}
+rule "charset" {
+  charset = "-"
+  min-chars = 1
+}
+```
+
+</details>
```
Comment (schavis) [inline website/content/docs/concepts/password-policies.mdx:393]:
```suggestion
```

File: website/content/docs/concepts/password-policies.mdx [hunk 5]
```diff
@@ -360,6 +360,39 @@ This policy generates 8 character passwords from the charset `abcde01234` and re
...
+## Default Password Policy
+
+Vault ships with a Default Password Policy and this gets applied to any password that Vault 
+generates where a custom password policy is not explicitly set. The default password policy is 
+defined below.
+
+<details>
+<summary>20 characters with at least 1 uppercase, 1 lowercase, 1 number, and 1 dash character.</summary>
+
+```hcl
```
Comment (schavis) [inline website/content/docs/concepts/password-policies.mdx:372]: I'm confused about what the HCL example is for. Are we expecting readers to copy it and use it somewhere in their config?

**Review by schavis — CHANGES_REQUESTED** [review]

File: website/content/docs/concepts/password-policies.mdx (hunk 5 above)
Comment (ramnarayanp) [reply website/content/docs/concepts/password-policies.mdx:372]: The HCL extract here is the HCL representation of the default password policy and I added it merely as a reference. Happy to remove it!

Comment (ram-parameswaran) [reply website/content/docs/concepts/password-policies.mdx:377]: I added the HCL equivalent of the default password policy merely as a reference for customers wanting to write their own, but i guess a reference already exists [here](https://developer.hashicorp.com/vault/docs/concepts/password-policies#password-policy-syntax). Happy to remove it if it seems redundant.

Comment (schavis) [reply website/content/docs/concepts/password-policies.mdx:376]: I'm not against keeping it :)

**Review by schavis — APPROVED** [review]

**Comment by schavis** [comment]
Apologies for the delay in getting this merged and published.

**Pull request merged**